ingestion/src/metadata/ingestion/source/database/redshift/queries.py

```diff
@@ -0,0 +1,194 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+SQL Queries used during ingestion
+"""
+
+import textwrap
+
+# Not able to use SYS_QUERY_HISTORY here. Few users not getting any results
+REDSHIFT_SQL_STATEMENT = textwrap.dedent(
+    """
+  WITH
+  queries AS (
+    SELECT *
+      FROM pg_catalog.stl_query
+     WHERE userid > 1
+          {filters}
+          -- Filter out all automated & cursor queries
+          AND querytxt NOT LIKE '/* {{"app": "OpenMetadata", %%}} */%%'
+          AND querytxt NOT LIKE '/* {{"app": "dbt", %%}} */%%'
+          AND aborted = 0
+          AND starttime >= '{start_time}'
+          AND starttime < '{end_time}'
+          LIMIT {result_limit}
+  ),
+  full_queries AS (
+    SELECT
+          query,
+          LISTAGG(CASE WHEN LEN(RTRIM(text)) = 0 THEN text ELSE RTRIM(text) END, '')
+            WITHIN GROUP (ORDER BY sequence) AS query_text
+      FROM pg_catalog.stl_querytext
+      WHERE sequence < 327	-- each chunk contains up to 200, RS has a maximum str length of 65535.
+    GROUP BY query
+  ),
+  raw_scans AS (
+    -- We have one row per table per slice so we need to get rid of the dupes
+    SELECT distinct query, tbl
+      FROM pg_catalog.stl_scan
+  ),
+  scans AS (
+  	SELECT DISTINCT
+  		query,
+  		sti.database AS database_name,
+      sti.schema AS schema_name
+  	  FROM raw_scans AS s
+          INNER JOIN pg_catalog.svv_table_info AS sti
+            ON (s.tbl)::oid = sti.table_id
+  )
+  SELECT DISTINCT
+        q.userid,
+        s.query AS query_id,
+        RTRIM(u.usename) AS user_name,
+        fq.query_text,
+        s.database_name,
+        s.schema_name,
+        q.starttime AS start_time,
+        q.endtime AS end_time,
+        q.aborted AS aborted
+    FROM scans AS s
+        INNER JOIN queries AS q
+          ON s.query = q.query
+        INNER JOIN full_queries AS fq
+          ON s.query = fq.query
+        INNER JOIN pg_catalog.pg_user AS u
+          ON q.userid = u.usesysid
+    ORDER BY q.endtime DESC
+"""
+)
+
+
+REDSHIFT_GET_ALL_RELATION_INFO = textwrap.dedent(
+    """
+    SELECT
+      c.relname as name,
+      c.relkind
+    FROM pg_catalog.pg_class c
+    LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace     
+    WHERE c.relkind = 'r'
+      AND n.nspname = :schema
+    UNION
+    SELECT
+        tablename as name,
+        'e' as relkind
+    FROM svv_external_tables
+    WHERE schemaname = :schema;
```

Comment:
only query the required schema every time, instead of querying all and leaving the filter later as in the default dialect implementation